packages/leancode_hooks/lib/src/use_cooldown.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+/// Signature for function that is returned from [useCooldown].
+typedef CooledDown = void Function(VoidCallback callback);
+
+/// Limits function calls to once per [duration].
+///
+/// When [CooledDown] function that this hook returns is called, a cooldown of
+/// [duration] is started. During it, any calls to [CooledDown] function are
+/// ignored.
+CooledDown useCooldown(
+  Duration duration, [
+  List<Object?> keys = const <Object>[],
+]) {
+  final debouncer = useMemoized(() => _Cooldowner(duration), keys);
```

Comment:
By the way, does the `_Cooldowner` ever need to be recreated? What do you think about dropping `keys` as parameter to `useCooldown` altogether and just pass `[duration]` as a single key to `useMemoized`?

Reply:
It does not. I think that's a good idea, thank you :)